src/test/java/com/amazon/ion/conformance/expectations.kt

```diff
@@ -186,7 +188,6 @@ private fun TestCaseSupport.denotesInt(expectation: AnyElement, reader: IonReade
     assertFalse(reader.isNullValue, createFailureMessage(expectation))
     when (expectedValue.integerSize) {
         IntElementSize.LONG -> {
-            assertNotEquals(IntegerSize.BIG_INTEGER, reader.integerSize, createFailureMessage(expectation))
```

Comment:
🗺️ This could go either way. Either there's a bug in `ion-java` because integer size is determined by the number of bytes in the integer encoding rather than by the actual value of the integer, or it doesn't actually matter, so there's a bug here. In practice, this seems to happen for overpadded integer values, so I've chosen to leave the `ion-java` behavior the way it is and fix it here.

Reply:
I think that's sensible. ion-java's behavior is the result of a conscious decision to limit the complexity of `IonReader.getIntegerSize()`. This can always be revisited if we have a compelling reason to, although a better solution is probably "don't overpad integers if you want them to be efficient."